python/lsst/daf/butler/registries/sqlRegistry.py

```diff
@@ -999,4 +1022,9 @@ def _queryMetadata(self, element, dataId, columns):
         ).fetchone()
         if row is None:
             raise LookupError(f"{element.name} entry for {dataId} not found.")
-        return {c.name: row[c.name] for c in cols}
+        result = {c.name: row[c.name] for c in cols}
+        if element.name == "instrument":
+            self._cachedInstrumentEntries[dataId["instrument"]] = result
```

Comment:
One late comment - I'd probably do something like

    self._cachedInstrumentEntries.setdefault(dataId["instrument"], {}).update(result)

here (and two lines below) to keep existing columns in case there are some non-overlapping columns in requests.

Reply:
Good idea!